src/benchmark.cc

```diff
@@ -877,14 +882,17 @@ void ParseCommandLineFlags(int* argc, const char** argv) {
                        &FLAGS_benchmark_repetitions) ||
         ParseStringFlag(argv[i], "benchmark_format",
                         &FLAGS_benchmark_format) ||
-        ParseBoolFlag(argv[i], "color_print",
-                       &FLAGS_color_print) ||
+        ParseStringFlag(argv[i], "benchmark_color",
+                        &FLAGS_benchmark_color) ||
+        // Legacy flag for benchmark_color.
+        ParseStringFlag(argv[i], "color_print",
+                        &FLAGS_benchmark_color) ||
         ParseInt32Flag(argv[i], "v", &FLAGS_v)) {
       for (int j = i; j != *argc; ++j) argv[j] = argv[j + 1];
 
       --(*argc);
       --i;
-    } else if (IsFlag(argv[i], "help")) {
+    } else if (IsFlag(argv[i], "help") || HasBenchmarkFlagPrefix(argv[i])) {
```

Comment:
$ test/benchmark_test --benchmark_format=foo
benchmark [--benchmark_list_tests={true|false}]
          [--benchmark_filter=<regex>]
          [--benchmark_min_time=<min_time>]
          [--benchmark_repetitions=<num_repetitions>]
          [--benchmark_format=<tabular|json|csv>]
          [--color_print={true|false}]
          [--v=<verbosity>]

this is working as intended. the checks for specific args come next.